### PR TITLE
Refactors TX watchers

### PR DESCRIPTION
**Pull request by Moejoe90**
This refactors the txWatchers to sagas and gets rid of event channel in order to use sagas completely. 

This will give us the chance to use API calls

### CODE REVIEW

File: packages/web/app/modules/tx/event-channel/sagas.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+export function* getTransactionOrThrow(
+  { web3Manager }: TGlobalDependencies,
+  txHash: string,
+): Iterator<any> {
+  const tx: Web3.Transaction = yield web3Manager.getTransactionByHash(txHash);
+  const txReceipt: Web3.TransactionReceipt | null = yield web3Manager.internalWeb3Adapter.getTransactionReceipt(
+    txHash,
+  );
```
Comment (drakkein) [inline packages/web/app/modules/tx/event-channel/sagas.ts]: Maybe we should use `all` here as a part of improvements? 
```const { tx, txReceipt} = all({...})```

File: packages/web/app/modules/tx/event-channel/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  // Both requests `getTx` and `getTransactionReceipt` can end up in two seperate nodes
+  const isMined = tx && tx.blockNumber && txReceipt && txReceipt.blockNumber;
+
+  if (!isMined) {
+    return null;
+  }
+
+  if (txReceipt!.status === TRANSACTION_STATUS.REVERTED) {
+    if (txReceipt!.gasUsed === tx.gas) {
```
Comment (drakkein) [inline packages/web/app/modules/tx/event-channel/sagas.ts]: Do we need these non null assertions here? Isn't it handled by isMined? Probably ts gonna have problem resolving it, so it's not important.

File: packages/web/app/modules/tx/event-channel/types.ts [hunk 3]
```diff
@@ -15,17 +13,16 @@ export type TEventEmitterChannelEvents =
     }
   | {
       type: EEventEmitterChannelEvents.TX_MINED;
-      tx: Web3.Transaction;
     }
   | {
       type: EEventEmitterChannelEvents.ERROR;
-      error: any;
+      error: Error;
```
Comment (drakkein) [inline packages/web/app/modules/tx/event-channel/types.ts]: <3

File: packages/web/app/modules/tx/monitor/sagas.ts [hunk 4]
```diff
@@ -135,11 +135,7 @@ export function* ensumrePendingTransactionSchemaIsValid(
   }
 }
 
-export function* updatePendingTxs({
-  apiUserService,
-  web3Manager,
-  logger,
-}: TGlobalDependencies): any {
+export function* updatePendingTxs({ apiUserService, logger }: TGlobalDependencies): any {
```
Comment (drakkein) [inline packages/web/app/modules/tx/monitor/sagas.ts]: Iterator<any>, also in line 209

Comment (Moejoe90) [reply packages/web/app/modules/tx/monitor/sagas.ts]: done

File: packages/web/app/modules/tx/event-channel/sagas.ts (hunk 2 above)
Comment (Moejoe90) [reply packages/web/app/modules/tx/event-channel/sagas.ts]: The TSC complains here if i remove the non null assertions, I think there is still some work to be done there. So i will leave them for now

Comment (AlexeyTuboltsev) [reply packages/web/app/modules/tx/event-channel/sagas.ts]: btw if you'd write it like this:
```
if (!(tx && tx.blockNumber && txReceipt && txReceipt.blockNumber)) {
    return null;
  } else if (txReceipt.status === TRANSACTION_STATUS.REVERTED) {
    if (txReceipt.gasUsed === tx.gas) {
      // All gas is burned in this case
      throw new OutOfGasError();
    }
    throw new RevertedTransactionError();
  } else {
    return tx;
  }
```
there won't be any problems with TSC (and it makes readability a bit better, but this is a matter of taste)

Comment (AlexeyTuboltsev) [reply packages/web/app/modules/tx/event-channel/sagas.ts]: btw if you'd write it like this:
```
if (!(tx && tx.blockNumber && txReceipt && txReceipt.blockNumber)) {
    return null;
  } else if (txReceipt.status === TRANSACTION_STATUS.REVERTED) {
    if (txReceipt.gasUsed === tx.gas) {
      // All gas is burned in this case
      throw new OutOfGasError();
    }
    throw new RevertedTransactionError();
  } else {
    return tx;
  }
```
there won't be any problems with TSC (and it makes readability a bit better, but this is a matter of taste)

Comment (AlexeyTuboltsev) [reply packages/web/app/modules/tx/event-channel/sagas.ts]: ok, let's leave it like this to preserve the `isMined` constant for readability

**Pull request merged**